File: C++/Guarana.cpp
#include "Guarana.h"

Guarana::Guarana(Swiat& swiat, int x, int y) :
	Roslina(swiat, x, y)
{
	this->znak = GUARANA;
}

Guarana::~Guarana()
{
}

bool Guarana::kolizja(Organizm* other)
{
	other->setSila(other->getSila() + 3);
	return true;
}

File: Java/Guarana.java
import java.awt.*;

public class Guarana extends Roslina{

    public Guarana(Swiat swiat, int x, int y) {
        super(swiat, x, y);
        this.znak = TypOrganizmu.GUARANA;
        setKolor(Color.RED);
    }
    boolean kolizja(Organizm other)
    {
        other.setSila(other.getSila() + 3);
        return true;
    }
}
